bucketlist/views.py

```diff
@@ -0,0 +1,223 @@
+    except:
+        return jsonify({"Message": "There are no bucketlists matching your request. Please try again"}), 404
+
+    if bucketlists.has_next:
+        next_page = "/bucketlists/?" + "limit=" + str(limit) + "&page=" + str(page + 1)
+    else:
+        next_page = "None"
+    if bucketlists.has_prev:
+        prev_page = "/bucketlists/?" + "limit=" + str(limit) + "&page=" + str(page - 1)
+    else:
+        prev_page = "None"
+
+    return jsonify({"count": len(bucketlists.items),
+                    "next": next_page,
+                    "prev": prev_page,
+                    "Bucketlists": [bucketlist.export_data() for bucketlist in bucketlists.items]}), 200
+
+
+@app.route("/bucketlists/<int:bucket_id>", methods=["GET"])
+@auth_token.login_required
+def get_bucketlist(bucket_id):
+    """
+    Returns a specified bucketlist.
+    """
+    # ensures that a logged-in user can only edit their own bucketlist
+    bucketlist = Bucketlist.query.filter_by(id=bucket_id, created_by=g.user.id).first()
+    if not bucketlist:
+        return jsonify({"Message": "That bucketlist does not exist for your user account. Please try again"}), 404
+    return jsonify({"Bucketlist": bucketlist.export_data()}), 200
+
+
+@app.route("/bucketlists/<int:bucket_id>", methods=["PUT"])
+@auth_token.login_required
+def update_bucketlist(bucket_id):
+    """
+    Updates a specified bucketlist.
+    """
+    # ensures that a logged-in user can only edit their own bucketlist
+    bucketlist = Bucketlist.query.filter_by(id=bucket_id, created_by=g.user.id).first()
+    if not bucketlist:
+        return jsonify({"Message": "That bucketlist does not exist for your user account. Please try again"}), 404
+    updated = bucketlist.update_data(request.json)
+    db.session.commit()
+    return jsonify({"Message": "Updated to " + updated.name.title()}), 200
+
+
+@app.route("/bucketlists/<int:bucket_id>", methods=["DELETE"])
+@auth_token.login_required
+def delete_bucketlist(bucket_id):
+    """
+    Deletes a specified bucketlist.
+    """
+    # ensures that a logged-in user can only edit their own bucketlist
+    bucketlist = Bucketlist.query.filter_by(id=bucket_id, created_by=g.user.id).first()
+    if not bucketlist:
+        return jsonify({"Message": "That bucketlist does not exist for your user account. Please try again"}), 404
+    db.session.delete(bucketlist)
+    db.session.commit()
+    return jsonify({"Message": bucketlist.name.title() + " has been deleted"}), 200
+
+
+@app.route("/bucketlists/<int:bucket_id>/items", methods=["POST"])
+@auth_token.login_required
+def new_item(bucket_id):
+    """
+    Creates a new bucketlist item.
+    """
+    # check if bucketlist exists, then validate user input using try-catch block
+    bucketlist = Bucketlist.query.filter_by(id=bucket_id, created_by=g.user.id).first()
+    if not bucketlist:
+        return jsonify({"Message": "You can only create an item within your own bucketlist. Please try again"}), 404
```

Comment:
For an API the message needs to be descriptive and give full info but not too descriptive to the point that it exposes your api. Consider messages like 'permission denied', 'unauthorized access', 'resource not found' etc